lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1162,22 +1220,27 @@ private ObjectData buildObjectData(
         }
         if (pp.objectHighlight) {
             for(final String key : highlight.keySet()) {
-                String newKey = key;
-                if (key.startsWith("key.")) {
-                    newKey = stripKeyPrefix(key);
-                }
-                b.withHighlight(newKey, highlight.get(key));
+                b.withHighlight(getReadableKeyNames(key), highlight.get(key));
             }
         }
 
         return b.build();
     }
 
+    private String getReadableKeyNames(final String key) throws IllegalStateException{
+        if (key.startsWith("key.")) {
+            return stripKeyPrefix(key);
+        } else if(readableNames.containsKey(key)) {
+            return readableNames.get(key);
+        } else {
+            //this should not happen. Untested
+            throw new IllegalStateException("Object has unexpected key");
```

Comment:
Include the offending key in the error message. It'd be nice if it included the guid as well but we can leave that for later